### PR TITLE
Added profile backend

### LINKED ISSUE #11: Create Profile backend
Create models for profile. 
Tasks - 
- [x] create a profile on the creation of the user.
- [x] send details through ``profile view`` by context.

Fields to be added-
user, coins, gender, DOB, total play count, won matches, draw matches, lost matches,

### PR DESCRIPTION
**Pull request by alexdeathway**
Created models and required fields as  mentioned in #11, also fixed issue with account/migration which was getting ignored by **manage.py makemigrations** and **migrate**.

### CODE REVIEW

**Comment by alexdeathway** [comment]
Backend is ready but is not integrated with the frontend for modifying Profile Model values.

**Comment by niteshsinha17** [comment]
@alexdeathway Nice work. 
Please read further comments.

File: account/models.py [hunk 1]
```diff
@@ -1,8 +1,32 @@
...
 
 # Create your models here.
 
 
 # class User(AbstractUser):
 #     guest = models.BooleanField(default=False)
+
+from django.contrib.auth.models import User
+
+class ProfileModel(models.Model):
```
Comment (niteshsinha17) [inline account/models.py:13]: Name it just ``Profile``

Comment (niteshsinha17) [inline account/models.py]: Please name it as ``Profile``

File: account/models.py [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
+       ('F',"Female"),
+       ('O',"Other"),
+       ('P',"Prefer not to say"),
+    )
+
+    user=models.OneToOneField(User,on_delete=models.CASCADE,related_name="Profile")
+    coins=models.PositiveIntegerField(default=0)
+    gender=models.CharField(max_length=1,choices=GENDER_CHOICES)
+    DOB=models.DateField(default=timezone.now)
+    TotalPlayCount=models.PositiveIntegerField(default=0)
```
Comment (niteshsinha17) [inline account/models.py]: Don't capitalize fields name. name it as ``dob``, ``total_play_count``, etc.

File: account/models.py [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+    MatchesWon=models.PositiveIntegerField(default=0)
+    MatchesDraw=models.PositiveIntegerField(default=0)
+    MatchesLost=models.PositiveIntegerField(default=0)
+
+
+    def __str__(self):
+        return f'{self.user} Profile'
```
Comment (niteshsinha17) [inline account/models.py]: It needs a new line

File: account/signals.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+@receiver(post_save,sender=User)
+def create_profile(sender,instance,created,**kwargs):
+    if created:
+        ProfileModel.objects.create(user=instance)
+
+
+@receiver(post_save,sender=User)
+def save_profile(sender,instance,**kwargs):
+    instance.Profile.save()
```
Comment (niteshsinha17) [inline account/signals.py]: It needs a new line

File: account/views.py [hunk 5]
```diff
@@ -51,6 +52,13 @@ def register(request):
     return render(request, "account/register.html", context)
 
 
+def profile(request):
```
Comment (niteshsinha17) [inline account/views.py]: It Needs Authentication check.

File: gamezilla/settings.py [hunk 6]
```diff
@@ -37,7 +37,7 @@
     "django.contrib.sessions",
     "django.contrib.messages",
     "django.contrib.staticfiles",
-    "account",
+    "account.apps.AccountConfig",
```
Comment (niteshsinha17) [inline gamezilla/settings.py:40]: Can't we use here only ``account``? Please confirm.

File: gamezilla/urls.py [hunk 7]
```diff
@@ -39,7 +40,7 @@
     # You MUST use an empty string as the URL prefix
     path("", include("pwa.urls")),
     path("new-home/",game_views.new_home),
-    path("profile/",game_views.profile)
```
Comment (niteshsinha17) [inline gamezilla/urls.py:42]: Please remove this view from ``profile`` app

**Review by niteshsinha17 — CHANGES_REQUESTED** [review]

File: account/models.py (hunk 3 above)
Comment (alexdeathway) [reply account/models.py]: new line?

File: gamezilla/settings.py (hunk 6 above)
Comment (alexdeathway) [reply gamezilla/settings.py:40]: Here <app_name>Config is required for signals (In this case for creating profile when user register), not sure if signal configuration with only  <app_name> in INSTALLED_APPS is possible, documentation suggest using this way .

File: account/views.py (hunk 5 above)
Comment (alexdeathway) [reply account/views.py]: My bad! will fix.

File: gamezilla/urls.py (hunk 7 above)
Comment (alexdeathway) [reply gamezilla/urls.py:42]: 25//`from account import views as game_views`

42//`path("profile/",game_views.profile)`

like this?

Comment (niteshsinha17) [reply gamezilla/urls.py:42]: No, 
Actually, you need to remove this view.
https://github.com/niteshsinha17/GameZilla/blob/b97854b38a5f5b98f1bb439fdcf530d5722b6a4d/game/views.py#L26-L28

File: account/models.py (hunk 2 above)
Comment (niteshsinha17) [reply account/models.py]: Here, don't set the default value of ``dob``. keep it ``blank=True`` and ``null=True``

File: account/models.py (hunk 3 above)
Comment (niteshsinha17) [reply account/models.py]: You can ignore it for now. But by a new line, I mean that we should end the code with a new line like this.
```
def __str__(self):
        return f'{self.user} Profile'
//space
```
Don't write ``//space``. It's all about space only. 😄  GitHub is also showing a red 🔴 mark there.

**Comment by niteshsinha17** [comment]
It's almost ready to get merged. Some changes are left.

File: account/models.py (hunk 3 above)
Comment (alexdeathway) [reply account/models.py]: lol! never noticed  github's that feature xd.

**Review by alexdeathway — COMMENTED** [review]
okay

File: account/models.py (hunk 2 above)
Comment (alexdeathway) [reply account/models.py]: fine

File: account/models.py (hunk 1 above)
Comment (alexdeathway) [reply account/models.py]: Done

**Comment by alexdeathway** [comment]
@niteshsinha17 `review requested`

**Comment by niteshsinha17** [comment]
Good Work 👏

**Review by niteshsinha17 — APPROVED** [review]

**Pull request merged**